subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -799,6 +960,19 @@ static bool send_request(enum http_method method, enum hawkbit_http_request type
 	const char *exec = hawkbit_status_execution(execution);
 	char device_id[DEVICE_ID_HEX_MAX_SIZE] = { 0 };
 #ifndef CONFIG_HAWKBIT_DDI_NO_SECURITY
+#ifdef CONFIG_HAWKBIT_SET_SETTINGS_RUNTIME
+#ifdef CONFIG_HAWKBIT_DDI_GATEWAY_SECURITY
+	static const char header_auth[] = "Authorization: GatewayToken ";
+#else
+	static const char header_auth[] = "Authorization: TargetToken ";
+#endif /* CONFIG_HAWKBIT_DDI_GATEWAY_SECURITY */
+
+	size_t s_header = strlen(header_auth) + DDI_SECURITY_TOKEN_SIZE + strlen("\r\n") + 1;
+	char *header = (char *)malloc(s_header);
```

Comment:
don't think this is ever free()'d?

Reply:
it is now changed to a array, so we don’t need free()